### ISSUE TITLE
Chapter 9/security - undefined variables in examples

### ISSUE THREAD

**Issue by Met7** [ethereumbook/ethereumbook#952]
Example 3 (original and fixed version), uses "balance" in withdraw(). Example 5 uses "transferValue" in claimReward(). Both undefined. The variables seem necessary to avoid reentrancy, just the definition is missing. Example 5 is where I got so far. I did a quick scan of the rest of the chapter and seems ok but can't say for sure.

**Comment by YannyD** [comment]
Yes example 5 does have several errors. Additionally, immediately following the example it refers to "depositedEther" rather than "depositedWei". 

I also see an issue in the explanation of example 9-7. He writes that line 21 calls fibonacci(uint256) but it actually is calling setFibonacci. 

I haven't had issues with the book this far, but this chapter seems a bit sloppy. I also wonder why they placed it so early in the book...
